File: 여행경로/jinbekim.js
function solution(tickets) {
	var answer = [];
	const DEP = 0,
		ARR = 1;
	DFS(tickets, 'ICN', ['ICN']);
	return answer.sort()[0];
	function DFS(tickets, depature, route) {
		if (tickets.length == 0) {
			answer.push(route);
		}
		for (var i in tickets) {
			if (tickets[i][0] == depature) {
				let tmp = tickets.slice();
				tmp.splice(i, 1);
				let tmp2 = route.slice();
				tmp2.push(tickets[i][ARR]);
				DFS(tmp, tickets[i][ARR], tmp2);
			}
		}
	}
}

console.log(
	solution([
		['ICN', 'SFO'],
		['ICN', 'ATL'],
		['SFO', 'ATL'],
		['ATL', 'ICN'],
		['ATL', 'SFO'],
	])
);

File: 전화번호목록/jinbekim.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

bool solution(vector<string> phone_book)
{
    bool answer = true;
    sort(phone_book.begin(), phone_book.end());
    for (int i = 0, len = phone_book.size(); i < len - 1; ++i)
    {
        int j = 0;
        while (phone_book[i][j] && phone_book[i + 1][j])
        {
            if (phone_book[i][j] != phone_book[i + 1][j])
                break;
            j++;
        }
        if (phone_book[i].length() == j || phone_book[i + 1].length() == j)
            return false;
    }
    return answer;
}